### ISSUE TITLE
`deprecated` decorator doesn't work on context managers

### ISSUE THREAD

**Issue by DetachHead** [microsoft/pyright#5949]
**Code or Screenshots**

```python
from contextlib import contextmanager

from typing_extensions import deprecated

@contextmanager
@deprecated("asdf")
def foo():
    yield

@deprecated("asdf")
@contextmanager
def bar():
    yield

with foo(): ... # no deprecation warning
with bar(): ... # no deprecation warning

```

**VS Code extension or command-line**
pylance 2023.9.10

**Comment by erictraut** [comment]
The `@deprecated` must be applied to a function, class or method. It won't work if applied to an object, like a context manager instance. See PEP 702 for details.

**Comment by DetachHead** [comment]
@erictraut the `@contextmanager` decorator does not return an instance of a context manager, it returns a function that returns a context manager, so the `@deprecated` decorator should still work

```py
def contextmanager(func):
    @wraps(func)
    def helper(*args, **kwds):
        return _GeneratorContextManager(func, args, kwds)
    return helper
```

**Comment by erictraut** [comment]
Ah yes, I misdiagnosed it. 

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.328, which I just published. It will also be included in a future release of pylance.
